app/models/tag.rb

```diff
@@ -1,7 +1,12 @@
 class Tag < ActiveRecord::Base
   include Searchable
 
-  validates :name, presence: true, uniqueness: true
+  validates :name, presence: true, uniqueness: { case_sensitive: false }
+
+  scope :with_name, ->(name) do
+    name = Array(name).map(&:downcase)
```

Comment:
Nitpick: since `name` is an array, shouldn't it be called `names`?

Reply:
It can also take a single string - it works the same way as 

```
tag_name = "hello"
tag_name_array = ["hello", "world"]
Tag.where(name: tag_name)
Tag.where(name: tag_name_array)
```

What do you think about `Tag.where_name`? That would probably be more clear.